File: Vachagan/C/vigenere.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>

int main(int argc, string argv[]){

    if (argc != 2){
        printf("Please enter valid [KEY], ex. ./vigenere abc \n");
        return 1;
    }
    
    string keyword = argv[1];
    // printf("%s\n", argv[]);
    int keylength = strlen(keyword);
    for (int i = 0; i < keylength; i++) {
        if(tolower(keyword[i]) < 'A' || tolower(keyword[i]) > 'z' || (tolower(keyword[i]) > 'Z' && tolower(keyword[i]) < 'a')){
            printf("Please enter valid [KEY], KEY only can alpha \n");
            return 1;
        }
    }

    string sentence = get_string("plaintext : ");

    int key = 0;
    int len = strlen(sentence);

    printf("ciphertext : ");
    for (int i = 0; i < len; i++) {
        if(sentence[i] >= 'A' && sentence[i] <= 'Z'){
            sentence[i] = (char)((sentence[i] - 'A' + tolower(keyword[key++])-'a')%26 + 'A');
        }else if (sentence[i] >= 'a' && sentence[i] <= 'z'){
            sentence[i] = (char)((sentence[i] - 'a' + tolower(keyword[key++])-'a')%26 + 'a');
        }
        printf("%c",sentence[i]);
        if (key == keylength)
            key = 0;
    }
    printf("\n");  
}

File: Vachagan/CS50x 2019/Week 6/vigenere/vigenere.py
from sys import argv
from cs50 import get_string
from itertools import cycle


def main(argv):

    if (len(argv) != 2):
        exit("Please enter valid [KEY], ex. ./vigenere abc")

    if not argv[1].isalpha():
        exit("Please enter valid [KEY], KEY only can alpha")

    sentence = get_string("plaintext: ")
    keyOrd = cycle([ord(lt) - 97 for lt in argv[1].lower()])
    sentence = ''.join([chr((ord(lt) % 32 + next(keyOrd)) % 26 + ord(lt) -
                            ord(lt) % 32) if lt.isalpha() else lt for lt in sentence])
    print("ciphertext:", sentence)

    # print("ciphertext: ", end='')
    # for lt in sentence:
    #     if lt.isalpha():
    #         case = ord(lt)%32
    #         lt = chr((case + next(keyOrd)) % 26 +   ord(lt) - case)
    #     print(lt, end='')
    # print()


if __name__ == "__main__":
    main(argv)
